cms/jinja2/templates/base.html

```diff
@@ -56,10 +57,42 @@
     }
 }
 %}
+
 {# fmt:on #}
 {% block head %}
     <link rel="stylesheet" href="{{ static('css/main.css') }}">
-{% endblock %}
+
+    {% if GOOGLE_TAG_MANAGER_CONTAINER_ID and not request.is_preview %}
+        <script>
+            function loadGTM(){
+                (function(w,d,s,l,i){
+                    w[l]=w[l]||[];
+                    w[l].push({
+                        'gtm.start': new Date().getTime(),
+                        event:'gtm.js'
+                    });
+                    var f=d.getElementsByTagName(s)[0],
+                        j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';
+                    j.async=true;
+                    j.src='https://www.googletagmanager.com/gtm.js?id='+i+dl;
+                    f.parentNode.insertBefore(j,f);
+                })
+                (window,document,'script','dataLayer', "{{ GOOGLE_TAG_MANAGER_CONTAINER_ID|escape }}");
+            }
+
+            function isUsageCookieAllowed(){
+                var re = /^(.*)?\s*'usage':true\s*[^;]+(.*)?$/;
+                return document.cookie.match(re);
+            }
+
+            if (isUsageCookieAllowed()) {
+                loadGTM();
+            }
```

Comment:
To verify this actually works, we should add a small integration test.

- Set cookie with name 'ons_cookie_policy' and value which contains ''usage':true'
- Assert in the head 'https://www.googletagmanager.com' and the `GOOGLE_TAG_MANAGER_CONTAINER_ID` is present.